device/esp_tinyusb/tinyusb_task.c

```diff
@@ -38,35 +39,61 @@ typedef struct {
     // TinyUSB stack configuration
     uint8_t rhport;                         /*!< USB Peripheral hardware port number. Available when hardware has several available peripherals. */
     tusb_rhport_init_t rhport_init;         /*!< USB Device RH port initialization configuration pointer */
+    const tinyusb_desc_config_t *desc_cfg;  /*!< USB Device descriptors configuration pointer */
     // Task related
     TaskHandle_t handle;                    /*!< Task handle */
-    TaskHandle_t awaiting_handle;           /*!< Task handle, waiting to be notified after successful start of TinyUSB stack */
+    volatile TaskHandle_t awaiting_handle;           /*!< Task handle, waiting to be notified after successful start of TinyUSB stack */
 } tinyusb_task_ctx_t;
 
-static tinyusb_task_ctx_t *p_tusb_task_ctx = NULL;
+static bool _task_is_running = false;               // Locking flag for the task, access only from the critical section
+static tinyusb_task_ctx_t *p_tusb_task_ctx = NULL;  // TinyUSB task context
 
 /**
  * @brief This top level thread processes all usb events and invokes callbacks
  */
 static void tinyusb_device_task(void *arg)
 {
+    tinyusb_task_ctx_t *task_ctx = (tinyusb_task_ctx_t *)arg;
+
+    // Sanity check
+    assert(task_ctx != NULL);
+    assert(task_ctx->awaiting_handle != NULL);
+
     ESP_LOGD(TAG, "TinyUSB task started");
-    assert(arg != NULL); // Sanity check
 
-    tinyusb_task_ctx_t *task_ctx = (tinyusb_task_ctx_t *)arg;
+    if (tud_inited()) {
+        ESP_LOGE(TAG, "TinyUSB stack is already initialized");
+        goto del;
+    }
+    if (tinyusb_descriptors_set(task_ctx->rhport, task_ctx->desc_cfg) != ESP_OK) {
+        ESP_LOGE(TAG, "TinyUSB descriptors set failed");
+        goto del;
+    }
     if (!tusb_rhport_init(task_ctx->rhport, &task_ctx->rhport_init)) {
         ESP_LOGE(TAG, "Init TinyUSB stack failed");
-        vTaskDelete(NULL);
-        // No return needed here: vTaskDelete(NULL) does not return
+        goto desc_free;
     }
+
     ESP_LOGD(TAG, "TinyUSB stack has been initialized in task");
```

Comment:
Since initializing TinyUSB in task is now the only option, could we remove this log?